prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,68 @@
+"""Taxi simulator program."""
+from prac_09.taxi import Taxi
+from prac_09.silver_service_taxi import SilverServiceTaxi
+
+MENU = "q)uit, c)hoose taxi, d)rive"
+
+
+def main():
+    """Display menu for taxi simulator program."""
+    print("Let's drive!")
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    current_taxi = None
+    total_fare = 0.00
+    print(MENU)
+    choice = input(">>> ").lower()
+    while choice != "q":
+        if choice == "c":
+            current_taxi = get_taxi(taxis, current_taxi)
+        elif choice == "d":
+            if current_taxi is None:
+                print("You need to choose a taxi before you can drive")
+            else:
+                drive_taxi(current_taxi)
+                total_fare += current_taxi.get_fare()
+        else:
+            print("Invalid option")
+        print(f"Bill to date: ${total_fare:.2f}")
+        print(MENU)
+        choice = input(">>> ").lower()
+    print(f"Total trip cost: {total_fare:.2f}")
+    print("Taxis are now:")
+    display_taxis(taxis)
+
+
+def display_taxis(taxis):
+    """Display taxis."""
+    for i, taxi in enumerate(taxis):
+        print(f"{i}, {taxi}")
+
+
+def get_taxi(taxis, current_taxi):
```

Comment:
Does this just get a taxi or does it get a valid taxi?

Reply:
I am getting a valid taxi here though I would consider that a side effect. You need to break out of the function if the user inputs something wrong. If you look at the solution it goes back to main on a wrong input. If I have a `get_valid_number` function then it loops until a valid number is entered but that's not what I  want. Usually I would use a `get_valid_number/taxi` function here but it seem unnecessary and went against the sample output.